modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/RefreshTokenHandler.java

```diff
@@ -0,0 +1,146 @@
+			requestParams.put(SCOPE_LABEL, PRODUCTION_LABEL);
+			APIUtilities apiUtilities = new APIUtilities();
+			apiUtilities.setEndPoint(IdentityProxy.getInstance().getAccessTokenURL());
+			apiUtilities.setHttpMethod(Constants.POST_METHOD);
+			apiUtilities.setRequestParamsMap(requestParams);
+			
+			byte [] credentials = Base64.encodeBase64((IdentityProxy.clientID + COLON + 
+					IdentityProxy.clientSecret).getBytes());
+			String encodedCredentials = new String(credentials);
+
+			Map<String, String> headers = new HashMap<String, String>();
+			
+			String authorizationString = Constants.AUTHORIZATION_MODE + encodedCredentials;
+			headers.put(Constants.AUTHORIZATION_HEADER, authorizationString);
+			headers.put(Constants.CONTENT_TYPE_HEADER, Constants.DEFAULT_CONTENT_TYPE);
+
+			Map<String, String> responseParams =
+					ServerUtilities.postDataAPI(apiUtilities, headers);
+
+			response = responseParams.get(Constants.SERVER_RESPONSE_BODY);
+			responseCode = responseParams.get(Constants.SERVER_RESPONSE_STATUS);
+			return response;
+		}
+
+		@SuppressLint("SimpleDateFormat")
+		@Override
+		protected void onPostExecute(String result) {
+
+			String refreshToken = null;
+			String accessToken = null;
+			int timeToExpireSecond = 3000;
+			IdentityProxy identityProxy = IdentityProxy.getInstance();
+			try {
+				JSONObject response = new JSONObject(result);
+
+				if (responseCode != null && responseCode.equals(Constants.REQUEST_SUCCESSFUL)) {
+					refreshToken = response.getString(Constants.REFRESH_TOKEN);
+					accessToken = response.getString(Constants.ACCESS_TOKEN);
+					timeToExpireSecond = Integer.parseInt(response.getString(Constants.EXPIRE_LABEL));
+
+					token.setRefreshToken(refreshToken);
+					token.setAccessToken(accessToken);
+
+					SharedPreferences mainPref =
+							IdentityProxy.getInstance()
+							             .getContext()
+							             .getSharedPreferences(Constants.APPLICATION_PACKAGE,
+							                                   Context.MODE_PRIVATE);
+					Editor editor = mainPref.edit();
+					editor.putString(Constants.REFRESH_TOKEN, refreshToken);
+					editor.putString(Constants.ACCESS_TOKEN, accessToken);
+
+					DateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy HH:mm:ss");
+					Date date = new Date();
+					long expiresIN = date.getTime() + (timeToExpireSecond * 1000);
+					Date expireDate = new Date(expiresIN);
+					String strDate = dateFormat.format(expireDate);
+					token.setDate(strDate);
+					editor.putString(Constants.DATE_LABEL, strDate);
+					editor.commit();
+
+					identityProxy.receiveNewAccessToken(responseCode, Constants.SUCCESS_RESPONSE, token);
+
+				} else if (responseCode != null) {
+					JSONObject responseBody = new JSONObject(result);
+					String error = responseBody.getString(Constants.ERROR_LABEL);
+					String errorDescription = responseBody.getString(Constants.ERROR_DESCRIPTION_LABEL);
+					Log.d(TAG, error);
```

Comment:
As it appears we're logging some error messages in this particular line. Is it right to log it as a debug message?